### PR TITLE
Replace AtomicInteger

### PR DESCRIPTION
**Pull request by chrisdembia**
This PR removes the AtomicInteger class with `std::atomic<int>`.

This is the first of multiple PRs for replacing pthreads with c++11 threading support.

The target branch for this PR is `feature_replace_pthreads`, a feature branch I made to gather all the PRs I'll be making.

---
This change is [Reviewable](https://reviewable.io/reviews/simbody/simbody/586)

### CODE REVIEW

**Comment by chrisdembia** [comment]
I've mostly completed the replacement of pthreads (all that's left is to delete the pthreads files for Windows). You can see all of these changes in this branch: https://github.com/simbody/simbody/compare/replace_pthreads

**Comment by chrisdembia** [comment]
All the tests pass; the red x is from codecov.

**Comment by sherm1** [comment]
:lgtm: -- really happy to see this dusty old code sent to the trash bin. Thanks, @chrisdembia. Merge any time.

---

Reviewed 7 of 10 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/simbody/simbody/586#-:-L0RMgh5raCq0-u76T-J:b-vkd1xj)*

File: SimTKcommon/tests/TestAtomicInteger.cpp [hunk 1]
```diff
@@ -1,152 +0,0 @@
...
-    ASSERT(a == 9);
-    a %= 7;
-    ASSERT(a == 2);
-    a <<= 2;
-    ASSERT(a == 8);
-    a >>= 1;
-    ASSERT(a == 4);
-}
-
-void testParallelExecution() {
```
Comment (chrisdembia) [inline SimTKcommon/tests/TestAtomicInteger.cpp]: Should I leave this test in?

**Comment by chrisdembia** [comment]
Thanks @sherm1 . I had one question for you.

**Comment by sherm1** [comment]
Ah, thanks for asking about that. Looking at that test I see it is doing more than just testing the basic functioning of AtomicInteger (for which we can assume std::atomic is already tested). It also tests its interaction with ParallelExecutor which might have its own problems. And those tests are hard to write so I think it would be better to leave that one in. No need for the one that tests basic functioning of operators though.

**Comment by chrisdembia** [comment]
@sherm1 I added the test back, but also had to comment out a section related to `operator*=()`, which is not supported by `std::atomic`. Ready for review again.

---

Review status: 10 of 11 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/simbody/simbody/586#-:-L0T9T8Cjio08unPQYxv:b-u21tly)*

**Comment by sherm1** [comment]
Thanks -- looks good. A few minor comments, PTAL.

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[SimTKcommon/tests/TestAtomic.cpp, line 9 at r3](https://reviewable.io:443/reviews/simbody/simbody/586#-L0TDIrjRUajkXpyunaJ:-L0TDIrjRUajkXpyunaK:b-vtyytk) ([raw file](https://github.com/simbody/simbody/blob/82894d43690b60586121113f130c4d847559ad45/SimTKcommon/tests/TestAtomic.cpp#L9)):*
> ```cpp
>  * Medical Research, grant U54 GM072970. See https://simtk.org/home/simbody.  *
>  *                                                                            *
>  * Portions copyright (c) 2008-12 Stanford University and the Authors.        *
> ```

BTW 12->17

---

*[SimTKcommon/tests/TestAtomic.cpp, line 25 at r3](https://reviewable.io:443/reviews/simbody/simbody/586#-L0TCyaLbf1muyRKXBC-:-L0TCyaLbf1muyRKXBC0:bua1nbn) ([raw file](https://github.com/simbody/simbody/blob/82894d43690b60586121113f130c4d847559ad45/SimTKcommon/tests/TestAtomic.cpp#L25)):*
> ```cpp
> 
> // This test originally tested the SimTK::AtomicInteger class template. When
> // SimTK::AtomicInteger was replaced with C++11 std::atomic, this test cased
> ```

BTW cased -> case

---

*[SimTKcommon/tests/TestAtomic.cpp, line 106 at r3](https://reviewable.io:443/reviews/simbody/simbody/586#-L0TD5i3pBwFeTkK-e_T:-L0TD5i3pBwFeTkK-e_U:b74ohmx) ([raw file](https://github.com/simbody/simbody/blob/82894d43690b60586121113f130c4d847559ad45/SimTKcommon/tests/TestAtomic.cpp#L106)):*
> ```cpp
>         ASSERT(index == -1024);
>     }
>     */
> ```

BTW better just to nuke this dead code I think.

There are trailing blanks on some of the blank lines above in case you feel like tidying those up.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/simbody/simbody/586#-:-L0TDMaXzx9_x2u0wd61:b5jztp8)*

**Comment by chrisdembia** [comment]
Ready for review again.

---

Review status: 10 of 11 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/simbody/simbody/586#-:-L0_VJUvOIOsG87v7VDC:bi676nx)*

**Comment by sherm1** [comment]
Thanks @chrisdembia. FYI: in Reviewable, if a reviewer starts a comment with "BTW" or "FYI" it means they don't need to review the response. You can close it yourself by responding with a comment that starts with "Done". Happy to re-review if you want, though!

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/simbody/simbody/586#-:-L0aIbL3yKj2XEH4iDqk:b-6gp0yy)*

**Comment by chrisdembia** [comment]
> FYI: in Reviewable, if a reviewer starts a comment with "BTW" or "FYI" it means they don't need to review the response. You can close it yourself by responding with a comment that starts with "Done". 

Ah thanks for teaching me the ways of Reviewable.

**Pull request merged**